docs/website/docs/development/architecture/how-odo-works.md

```diff
@@ -0,0 +1,1382 @@
+| `odo.dev/project-type `                | the application runtime, if available. Optional. Value is read in order from the `metadata.projectType` or `metadata.language` fields in the Devfile  | `spring`            |
+
+Notes:
+- Any additional annotations defined via the `components[].container.annotation.deployment` field will also be added to this resource.
+- All those annotations are also added to the underlying Pods managed by this Deployment.
+
+<details>
+<summary>Example</summary>
+
+<table>
+<thead>
+<tr>
+<td>Devfile</td>
+<td></td>
+<td>Kubernetes Deployment</td>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td>
+
+```yaml
+metadata:
+  # highlight-start
+  name: my-sample-java-springboot
+  projectType: spring
+  # highlight-end
+  language: java
+components:
+- name: tools
+  container:
+     annotation:
+        deployment:
+           # highlight-start
+           example.com/my-annotation: value-1
+           # highlight-end
+```
+</td>
+<td> => </td>
+<td>
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+   # highlight-next-line
+   name: my-sample-java-springboot-app
+   annotations:
+      # highlight-start
+      odo.dev/project-type: spring
+      example.com/my-annotation: value-1
+      # highlight-end
+spec:
+   template:
+      metadata:
+         # highlight-next-line
+         name: my-sample-java-springboot-app
+         annotations:
+            # highlight-start
+            odo.dev/project-type: spring
+            example.com/my-annotation: value-1
+            # highlight-end
+```
+</td>
+</tr>
+</tbody>
+</table>
+
+</details>
+
+#### Labels
+
+By default, `odo` adds the following labels to the Deployment:
+
+| Key                                    | Description                                                                                                                                          | Example Value       |
+|----------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------|---------------------|
+| `app`                                  | the application; always `app`                                                                                                                        | `app`               |
+| `app.kubernetes.io/instance`           | the component name                                                                                                                                   | `my-component-name` |
+| `app.kubernetes.io/managed-by`         | the tool used to create this resource; always `odo`                                                                                                  | `odo`               |
+| `app.kubernetes.io/managed-by-version` | the version of the odo binary used to create this resource                                                                                           | `v3.0.0-rc1`        |
```

Comment:
As this is an example, I would use an hypothetic (for the moment), but more current version in the future (hopefully ;) )

```suggestion
| `app.kubernetes.io/managed-by-version` | the version of the odo binary used to create this resource                                                                                           | `v3.0.0`        |
```